platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultConfigurationResolver.java

```diff
@@ -195,7 +190,7 @@ public ResolverResults resolveBuildDependencies(ResolveContext resolveContext) {
     }
 
     @Override
-    public ResolverResults resolveGraph(ResolveContext resolveContext) {
+    public ResolverResults resolveGraph(ResolveContext resolveContext, List<? extends ResolutionAwareRepository> repositories) {
```

Comment:
🤔 This seems to make more sense to me in the original, keeping the `RespositoriesSupplier` available to these resolvers fixed as a field.  The semantics of a `ConfigurationResolver` seem to me like it should be "something specific to a supply of repositories, that can resolve a configuration against only those repositories".

If you want to resolve against a different set of repos, I think it makes sense to use a different resolver, instead of re-calling the same resolver with a different set of repos.

Would this still allow for moving repository selection outside of the resolution codepath?  Maybe we should talk in person about this.

Reply:
Happy to talk about this in person. I agree a `Resolver` should be able to resolve something without needing to supply the repositories. 

To me, this just seems like a naming change and maybe splitting this class into two with  some kind of `ResolverFactory`

Reply:
Moved the repository selection into the resolver and outside of the configuration. We can do splitting and renaming in future changes